paddle/fluid/framework/ir/mkldnn/fc_elementwise_add_mkldnn_fuse_pass.cc

```diff
@@ -0,0 +1,246 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "paddle/fluid/framework/ir/mkldnn/fc_elementwise_add_mkldnn_fuse_pass.h"
+
+#include <functional>
+#include <memory>
+#include <tuple>
+
+#include "paddle/fluid/framework/ir/graph_traits.h"
+#include "paddle/fluid/framework/op_version_registry.h"
+#include "paddle/fluid/string/pretty_log.h"
+
+namespace paddle {
+namespace framework {
+namespace ir {
+
+FCResidualConnectionMKLDNNFusePass::FCResidualConnectionMKLDNNFusePass() {
+  AddOpCompat(OpCompat("fc"))
+      .AddInput("Input")
+      .IsTensor()
+      .End()
+      .AddInput("W")
+      .IsTensor()
+      .End()
+      .AddInput("Bias")
+      .IsTensor()
+      .End()
+      .AddOutput("Out")
+      .IsTensor()
+      .End()
+      .AddAttr("in_num_col_dims")
+      .IsNumGE(1)
+      .End();
+
+  AddOpCompat(OpCompat("elementwise_add"))
+      .AddInput("X")
+      .IsTensor()
+      .End()
+      .AddInput("Y")
+      .IsTensor()
+      .End()
+      .AddOutput("Out")
+      .IsTensor()
+      .End()
+      .AddAttr("axis")
+      .IsIntIn({-1, 0, 1})
+      .End();
+}
+
+GraphWithStats FCResidualConnectionMKLDNNFusePass::FuseFC(
+    const std::string& name_scope, const GraphWithStats& graph_with_stats,
+    bool asX) const {
+  GraphPatternDetector gpd;
+  auto pattern = gpd.mutable_pattern();
+  patterns::FCMKLDNN fc_pattern{pattern, name_scope};
+  auto fc_output = fc_pattern(
+      gpd.mutable_pattern()->NewNode("fc")->AsInput()->assert_is_op_input(
+          "fc", "Input"),
+      true);
+
+  patterns::ResidualElementwise elementwise_pattern{pattern, name_scope, asX};
+  elementwise_pattern(
+      fc_output,
+      pattern->NewNode(elementwise_pattern.residual_data_repr()),
+      "elementwise_add", asX);
+  fc_output->AsIntermediate();
+
+  int found_fc_count = 0;
+
+  auto handler = [&](const GraphPatternDetector::subgraph_t& subgraph,
+                     Graph* g) {
+    GET_IR_NODE_FROM_SUBGRAPH(fc_op, fc, fc_pattern);
+    GET_IR_NODE_FROM_SUBGRAPH(fc_input, input, fc_pattern);
+    GET_IR_NODE_FROM_SUBGRAPH(fc_weights, weights, fc_pattern);
+    GET_IR_NODE_FROM_SUBGRAPH(fc_output, output, fc_pattern);
+
+    GET_IR_NODE_FROM_SUBGRAPH(elementwise_op, elementwise_op,
+                              elementwise_pattern);
+    GET_IR_NODE_FROM_SUBGRAPH(residual_data, residual_data,
+                              elementwise_pattern);
+    GET_IR_NODE_FROM_SUBGRAPH(elementwise_out, elementwise_out,
+                              elementwise_pattern);
+
+    if (FindFuseOption(*fc_op, *elementwise_op) != FUSE_MKLDNN) return;
+
+    if (!IsReachable(g, residual_data, fc_output)) return;
+
+    if (HasFusedActivation(fc_op)) return;
```

Comment:
I don't think we need here so many blank lines.
```suggestion
    if (FindFuseOption(*fc_op, *elementwise_op) != FUSE_MKLDNN) return;
    if (!IsReachable(g, residual_data, fc_output)) return;
    if (HasFusedActivation(fc_op)) return;
```